tensorflow/python/keras/engine/network.py

```diff
@@ -547,20 +547,22 @@ def get_layer(self, name=None, index=None):
     """
     # TODO(fchollet): We could build a dictionary based on layer names
     # since they are constant, but we have not done that yet.
-    if index is not None:
+    if index is not None and name is not None:
+      raise ValueError('Provide only a layer name or a layer index.')
+    elif index is not None:
```

Comment:
Since raising exception ends function body, there is no need for `elif`:

```suggestion

    if index is not None:
```